UnitystationLauncher/ContentScanning/AssemblyTypeChecker.cs

```diff
@@ -0,0 +1,1211 @@
+        Action<string> logErrors)
+    {
+        info.Invoke($"{name}: Verifying IL...");
+        var sw = Stopwatch.StartNew();
+        var bag = new ConcurrentBag<VerificationResult>();
+
+
+        var UesParallel = false;
+
+        if (UesParallel)
+        {
+            var partitioner = Partitioner.Create(reader.TypeDefinitions);
+            Parallel.ForEach(partitioner.GetPartitions(Environment.ProcessorCount), handle =>
+            {
+                var ver = new Verifier(resolver);
+                ver.SetSystemModuleName(new AssemblyName(SystemAssemblyName));
+                while (handle.MoveNext())
+                {
+                    foreach (var result in ver.Verify(peReader, handle.Current, verifyMethods: true))
+                    {
+                        bag.Add(result);
+                    }
+                }
+            });
+        }
+        else
+        {
+            var ver = new Verifier(resolver);
+            //mscorlib
+            ver.SetSystemModuleName(new AssemblyName(SystemAssemblyName));
+            foreach (var Definition in reader.TypeDefinitions)
+            {
+                var Errors = ver.Verify(peReader, Definition, verifyMethods: true);
+                foreach (var Error in Errors)
+                {
+                    bag.Add(Error);
+                }
+            }
+        }
+
+        var loadedCfg = _config.Result;
+
+        var verifyErrors = false;
+        foreach (var res in bag)
+        {
+            if (loadedCfg.AllowedVerifierErrors.Contains(res.Code))
+            {
+                continue;
+            }
+
+            var formatted = res.Args == null ? res.Message : string.Format(res.Message, res.Args);
+            var msg = $"{name}: ILVerify: {formatted}";
+
+            if (!res.Method.IsNil)
+            {
+                var method = reader.GetMethodDefinition(res.Method);
+                var methodName = FormatMethodName(reader, method);
+
+                msg = $"{msg}, method: {methodName}";
+            }
+
+            if (!res.Type.IsNil)
+            {
+                var type = GetTypeFromDefinition(reader, res.Type);
+                msg = $"{msg}, type: {type}";
+            }
+
+
+            verifyErrors = true;
+            logErrors.Invoke(msg);
+        }
+
+        info.Invoke($"{name}: Verified IL in {sw.Elapsed.TotalMilliseconds}ms");
+
+        if (verifyErrors)
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    private static string FormatMethodName(MetadataReader reader, MethodDefinition method)
+    {
+        var methodSig = method.DecodeSignature(new TypeProvider(), 0);
+        var type = GetTypeFromDefinition(reader, method.GetDeclaringType());
+
+        return
+            $"{type}.{reader.GetString(method.Name)}({string.Join(", ", methodSig.ParameterTypes)}) Returns {methodSig.ReturnType} ";
+    }
+
+    [SuppressMessage("ReSharper", "BitwiseOperatorOnEnumWithoutFlags")]
```

Comment:
I'm not able to find this warning when searching online, so I'm not entirely sure what it means. But guessing off the name I would assume its the `if`-s here, would it be better to leave the warning on and just ignore it, instead of disabling it? Just so its easier to track